Document/0x05g-Testing-Network-Communication.md

```diff
@@ -117,6 +117,21 @@ Certificate pinning is the process of associating the backend server with a part
 
 The certificate can be pinned and hardcoded into the app or retrieved at the time the app first connects to the backend. In the latter case, the certificate is associated with ("pinned" to) the host when the host is seen for the first time. This alternative is less secure because attackers intercepting the initial connection can inject their own certificates.
 
+##### When the pin fails
+
+Note that there are various ways to work with a failing pin:
+
+- Inform the user about not being able to connect to the backend and stop all operations. The app can check whether there is an update and inform the user about updating to the latest version of the app if available. The app allows no longer for any form of interaction with the user until it is updated or the pin works again.
+- Do a call to a crash-reporting service including that the pinning has failed to notify the responsible developer(s) about a security misconfiguration.
+- The app calls the backend using a TLS enabled call with no pinning to inform the backend of a pinning failure. The call can either differ in user-agent, JWT token-contents, or have other headers with a flag enabled of pinning failure.
+- After calling the backend or crash-reporting service to notify about the failing pinning, the app can still offer limited functionality that shouldn't involve sensitive functions or processing of sensitive data. The communication would happen without SSL Pinning and just validate the X.509 certificate accordingly.
+
+Which option(s) you choose depends on how important availability is compared to the complexity of maintaining the application.
+
+When a large amount of pinfailures are reported to the backend or crash-reporting service, the developer should understand that there is probably a misconfiguration. There is a large chance that the keymaterials used at the TLS terminating endpoint (e.g. server/loadbalancer) is different than what the app is expecting. In that case an update of either that keymaterial or an update of the app should be pushed through.
+
+When only very few pinfailures are reported, then the network is sitll ok, and so is the configuration of the TLS terminating endpoint. Instead, it might well be that there is a Man in The Middle attack ongoing at the app instance of which the pin is failing.
```

Comment:
```suggestion
When only very few pin failures are reported, then the network should be ok, and so should be the configuration of the TLS terminating endpoint. Instead, it might well be that there is a man-in-the-middle attack ongoing at the app instance of which the pin is failing.
```